pkg/apis/batch/validation/validation.go

```diff
@@ -522,6 +578,21 @@ func validateJobStatus(job *batch.Job, fldPath *field.Path, opts JobStatusValida
 			}
 		}
 	}
+	if ptr.Deref(job.Spec.CompletionMode, batch.NonIndexedCompletion) == batch.NonIndexedCompletion && isJobSuccessCriteriaMet(job) {
```

Comment:
```suggestion
	if ptr.Deref(job.Spec.CompletionMode, batch.NonIndexedCompletion) != batch.IndexedCompletion && isJobSuccessCriteriaMet(job) {
```

If we add a new completionMode, this keeps validation as we currently intend it, correct?

Reply:
I'm addressing it.

Reply:
Done.